core/blockchain.go

```diff
@@ -2014,20 +2016,27 @@ func (bc *BlockChain) WritePendingCrossLinks(crossLinks []types.CrossLink) error
 
 }
 
-// AddPendingSlashingCandidate  appends pending slashing candidates
-func (bc *BlockChain) AddPendingSlashingCandidate(candidate *slash.Record) (int, error) {
+// AddPendingSlashingCandidates appends pending slashing candidates
+func (bc *BlockChain) AddPendingSlashingCandidates(
+	candidates []slash.Record,
+) (int, error) {
 	bc.pendingSlashingCandidatesMU.Lock()
 	defer bc.pendingSlashingCandidatesMU.Unlock()
-
 	cls, err := bc.ReadPendingSlashingCandidates()
+
 	if err != nil || len(cls) == 0 {
-		err := bc.WritePendingSlashingCandidates([]slash.Record{*candidate})
+		err := bc.WritePendingSlashingCandidates(candidates)
+		if err != nil {
+			return 0, err
+		}
 		return 1, err
 	}
-	cls = append(cls, *candidate)
-	err = bc.WritePendingSlashingCandidates(cls)
-	return len(cls), err
 
+	cls = append(cls, candidates...)
```

Comment:
no dedup?

Reply:
I think technically not needed since I clear out all the slashing candidates in end writeblockwithstate